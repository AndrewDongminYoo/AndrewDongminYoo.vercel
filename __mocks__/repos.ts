const username = 'AndrewDongminYoo';
const repoName = `${username}.vercel`;
const mockName = `${username}/${repoName}`;
const usersApi = `https://api.github.com/users/${username}`;
const reposApi = `https://api.github.com/repos/${mockName}`;
const mockRepository = {
    id: 447431678,
    node_id: 'R_kgDOGqtD_g',
    name: repoName,
    full_name: mockName,
    private: false,
    owner: {
        login: username,
        id: 82999715,
        node_id: 'MDQ6VXNlcjgyOTk5NzE1',
        avatar_url: 'https://avatars.githubusercontent.com/u/82999715?v=4',
        gravatar_id: '',
        url: `${usersApi}`,
        html_url: `https://github.com/${username}`,
        followers_url: `${usersApi}/followers`,
        subscriptions_url: `${usersApi}/subscriptions`,
        organizations_url: `${usersApi}/orgs`,
        repos_url: `${usersApi}/repos`,
        received_events_url: `${usersApi}/received_events`,
        type: 'User',
        site_admin: false,
    },
    html_url: `https://github.com/${mockName}`,
    description: '깃허브 메인 프로필 (+이력서)입니다. NextJS',
    fork: false,
    languages_url: `${reposApi}/languages`,
    created_at: '2022-01-13T02:03:50Z',
    updated_at: '2023-01-18T13:04:10Z',
    pushed_at: '2023-01-18T13:34:47Z',
    svn_url: `https://github.com/${mockName}`,
    homepage: `https://${username}.github.io`,
    size: 35801,
    stargazers_count: 2,
    watchers_count: 2,
    language: 'TypeScript',
    has_issues: true,
    has_projects: true,
    has_downloads: true,
    has_wiki: true,
    has_pages: true,
    has_discussions: false,
    forks_count: 0,
    archived: false,
    disabled: false,
    open_issues_count: 1,
    allow_forking: true,
    is_template: false,
    web_commit_signoff_required: false,
    topics: ['nextjs', 'profile', 'react', 'resume', 'tailwindcss'],
    visibility: 'public',
    forks: 0,
    open_issues: 1,
    watchers: 2,
    default_branch: 'main',
    languages: {
        TypeScript: 79630,
        CSS: 11761,
        JavaScript: 6169,
    },
};

export const readRepositories = jest.fn(() => [mockRepository, mockRepository, mockRepository]);
