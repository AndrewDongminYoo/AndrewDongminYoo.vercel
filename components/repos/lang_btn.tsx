import colorMap, { Language } from './lang_colors';
import Link from 'next/link';
import styles from '@styles/repository.module.css';

export default function LanguageButton({
    language,
    percent,
    index,
}: {
    language: Language;
    percent: number;
    index: number;
}) {
    const backgroundColor = colorMap[language] as string;
    const myRepoLanguage = `https://github.com/AndrewDongminYoo?tab=repositories&language=${language}`;
    const trendingOfLang = `https://github.com/topics/${language}`;
    const style = { backgroundColor };
    return (
        <li className={styles.language__component}>
            <Link href={index === 0 ? myRepoLanguage : trendingOfLang}>
                <span
                    className={styles.language_button}
                    style={style}
                    aria-hidden={true}
                />
                <span className={styles.language_name}>{language}</span>
                <span>{percent.toFixed(1) + '%'}</span>
            </Link>
        </li>
    );
}
