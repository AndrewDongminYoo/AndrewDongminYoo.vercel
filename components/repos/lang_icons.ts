import aws from '@/public/svg/stacks/aws-lambda.svg';
import expo from '@/public/svg/stacks/expo.svg';
import flask from '@/public/svg/stacks/flask.svg';
import flutter from '@/public/svg/stacks/flutter.svg';
import nextjs from '@/public/svg/stacks/nextjs.svg';
import openai from '@/public/svg/stacks/openai.svg';
import reactnative from '@/public/svg/stacks/react-native.svg';
import rn_firebase from '@/public/svg/stacks/react-native-firebase.svg';
import selenium from '@/public/svg/stacks/selenium.svg';
import spring from '@/public/svg/stacks/spring.svg';
import swagger from '@/public/svg/stacks/swagger.svg';
import typescript from '@/public/svg/stacks/typescript-icon.svg';
import webpack from '@/public/svg/stacks/webpack.svg';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const frameworks: Record<string, any> = {
  '52market.front': webpack,
  '52market.shop': swagger,
  'AndrewDongminYoo.vercel': nextjs,
  'do-it-react-native-clone': typescript,
  'flask-My-Small-Meal': flask,
  'flutter-proxy-voting-app': flutter,
  'my-first-react-native-chatting-app': expo,
  'openai-chatgpt-my-conversations': openai,
  'react-native-step-counter': reactnative,
  'Serverless-Framework-Crawlers': aws,
  'TIL-of-BootCamp-Flask': selenium,
  'TIL-of-BootCamp-Spring': spring,
  walking_tracker: rn_firebase,
  wegooliFriendApp: flutter,
};

export default frameworks;
