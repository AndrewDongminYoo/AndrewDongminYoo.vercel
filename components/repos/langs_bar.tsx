import colorMap, { Language } from './lang_colors';

export default function LanguageStateBar({
    languages,
    totalCount,
}: {
    languages: [Language, number][];
    totalCount: number;
}) {
    return (
        <span
            data-view-component={true}
            className="flex h-3 mt-1 overflow-hidden bg-transparent border-collapse"
        >
            {languages.map(([language, count], id) => {
                return (
                    <LanguageBarPart
                        language={language}
                        percent={(count / totalCount) * 100}
                        key={id}
                    />
                );
            })}
        </span>
    );
}

function LanguageBarPart({
    language,
    percent,
}: {
    language: Language;
    percent: number;
}) {
    const backgroundColor = colorMap[language] as string;
    const percentString = percent.toFixed(1) + '%';
    return (
        <span
            style={{
                backgroundColor: `${backgroundColor} !important`,
                width: percentString,
            }}
            itemProp="keywords"
            aria-label={`${language} ${percentString}`}
            data-view-component={true}
            className="border-collapse flex h-3 overflow-hidden bg-transparent ml-[2px] first:ml-0"
        />
    );
}
