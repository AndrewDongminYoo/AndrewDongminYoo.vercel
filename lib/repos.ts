import { Endpoints } from '@octokit/types';
import { Octokit } from '@octokit/core';
import type { Repository } from '@typings/repos';
import fs from 'fs';
import { parse } from 'html-metadata-parser';
import { parseISO } from 'date-fns';
import path from 'path';

const octokit = new Octokit({ auth: process.env.GITHUB_TOKEN });
const reposDirectory = path.join(process.cwd(), 'data/repos');

/** 깃허브에서 가져온 데이터 필터링 함수 */
// eslint-disable-next-line @typescript-eslint/no-explicit-any
const reclusiveFilter = (repo: { [x: string]: any }) => {
    Object.entries(repo).forEach(([key, value]) => {
        switch (typeof value) {
            case 'string': {
                if (value.endsWith('.git')) {
                    delete repo[key];
                } else if (value.includes('{')) {
                    delete repo[key];
                } else {
                    repo[key] = value;
                }
                break;
            }
            case 'object': {
                if (key === 'topics') {
                    repo[key] = value;
                } else if (value === null) {
                    delete repo[key];
                } else {
                    repo[key] = reclusiveFilter(value);
                }
                break;
            }
            case 'undefined': {
                delete repo[key];
                break;
            }
            default: {
                repo[key] = value;
                break;
            }
        }
    });
    return repo as Repository;
};

/** 깃허브 API 통해 유저의 리포지토리 리스트 전체 가져오기 */
const fetchRepositories = async () => {
    const EP_REPOS: keyof Endpoints = 'GET /user/repos';
    const repositories = await octokit
        .request(EP_REPOS, {
            type: 'public',
            per_page: 100,
            direction: 'desc',
            sort: 'pushed',
        })
        .then((value) => value.data);
    return repositories
        .filter((R) => !R.fork && R.size > 4000 && R.topics?.length !== 0)
        .map((repo) => reclusiveFilter(repo));
};

/** 깃허브 API 통해 리포지토리 데이터 가져오기 */
const fetchRepository = async (owner: string, repo: string) => {
    const EP_REPO: keyof Endpoints = 'GET /repos/{owner}/{repo}';
    return await octokit
        .request(EP_REPO, { owner, repo })
        .then((value) => value.data)
        .then((repo) => reclusiveFilter(repo));
};

/** 깃허브에서 가져온 리포지토리 데이터 로컬에 저장하기 */
const downloadJSON = async () => {
    const repositories = await fetchRepositories();
    const repositoryData = await Promise.all(
        repositories.map(async (repo) => {
            const { html_url, languages_url } = repo;
            const languages = await octokit
                .request({ url: languages_url })
                .then((res) => res.data);
            const meta_tags = await getTagsFromWebsite(html_url);
            return { ...repo, meta_tags, languages };
        })
    );
    repositoryData.forEach((json) => {
        const targetJsonPath = path.join(reposDirectory, `${json.name}.json`);
        fs.writeFile(
            targetJsonPath,
            JSON.stringify(json, null, 4),
            { flag: 'w' },
            (err) => {
                if (err) console.error(err);
            }
        );
    });
    return repositoryData.length;
};
/** 로컬 리포지토리 데이터 ID 리스트 읽기 */
function readReposIds() {
    // Get file names under /data/repos
    const fileNames = fs.readdirSync(reposDirectory);
    return fileNames.map((fileName) => {
        // Remove '.md' from file name to get id
        const repo = fileName.replace(/\.json$/, '');
        return { params: { repo } };
    });
}

/** 로컬 리포지토리 데이터 ID로 읽어오기 */
const readData = (repo: string) => {
    // Read JSON file as Repository
    const fullPath = path.join(reposDirectory, `${repo}.json`);
    const fileContents = fs.readFileSync(fullPath, 'utf8');
    return JSON.parse(fileContents) as Repository;
};

/** 로컬 리포지토리 리스트 데이터 읽어오기 */
const readRepositories = () => {
    const allReposData = readReposIds().map(({ params: { repo } }) =>
        readData(repo)
    );
    // Sort repos by date
    return allReposData.sort((a, b) => {
        if (parseISO(a.pushed_at) < parseISO(b.pushed_at)) {
            return 1;
        } else return -1;
    });
};

async function getTagsFromWebsite(url: string) {
    return await parse(url).then((metaData) => metaData.og);
}

export {
    downloadJSON,
    fetchRepositories,
    fetchRepository,
    readData,
    readReposIds,
    readRepositories,
};
