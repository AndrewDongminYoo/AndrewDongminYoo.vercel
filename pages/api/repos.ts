import { NextApiRequest, NextApiResponse } from 'next';
import { Endpoints } from '@octokit/types';
import { Octokit } from '@octokit/core';
import fs from 'fs';
import path from 'path';

export default async function repos(req: NextApiRequest, res: NextApiResponse) {
    const octokit = new Octokit({ auth: process.env.GITHUB_TOKEN });
    const repos: keyof Endpoints = 'GET /user/repos';
    const info: keyof Endpoints = 'GET /repos/{owner}/{repo}';
    const fetchRepositories = async () => {
        const repositories = await octokit
            .request(repos, {
                type: 'public',
                per_page: 100,
                direction: 'desc',
                sort: 'created',
            })
            .then((value) => value.data);
        return repositories
            .filter((_) => !_.fork && _.size > 5000 && _.topics?.length !== 0)
            .map((repo) => reclusiveFilter(repo));
    };

    const fetchRepository = async (owner: string, repo: string) => {
        return await octokit
            .request(info, { owner, repo })
            .then((value) => value.data)
            .then((repo) => reclusiveFilter(repo));
    };

    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    const reclusiveFilter = (repo: { [x: string]: any }) => {
        Object.entries(repo).forEach(([key, value]) => {
            switch (typeof value) {
                case 'string': {
                    if (value.endsWith('.git')) {
                        delete repo[key];
                    } else if (key === 'languages_url') {
                        repo[key] = value;
                    } else if (value.startsWith('https://api.github.com/')) {
                        delete repo[key];
                    } else {
                        repo[key] = value;
                    }
                    break;
                }
                case 'object': {
                    if (key === 'topics') {
                        repo[key] = value;
                    } else if (value === null) {
                        delete repo[key];
                    } else {
                        repo[key] = reclusiveFilter(value);
                    }
                    break;
                }
                case 'undefined': {
                    delete repo[key];
                    break;
                }
                default: {
                    repo[key] = value;
                    break;
                }
            }
        });
        return repo;
    };

    const downloadJSON = async () => {
        const repositoryData = await fetchRepositories();
        repositoryData.forEach((json) => {
            const reposDirectory = path.join(
                process.cwd(),
                'repos',
                `${json.name}.json`
            );
            fs.writeFile(
                reposDirectory,
                JSON.stringify(json, null, 4),
                { flag: 'w' },
                (err) => {
                    if (err) console.error(err);
                }
            );
        });
        return repositoryData.length;
    };

    const { query, method } = req;
    if (method === 'GET') {
        if (typeof query.full_name === 'string') {
            const [owner, repo] = query.full_name.split('/');
            const repository = await fetchRepository(owner, repo);
            res.status(200).json({ repository });
        } else {
            const repositories = await fetchRepositories();
            res.status(200).json({ repositories });
        }
    } else if (method === 'POST') {
        await downloadJSON().then((length) => {
            res.status(200).send({ length });
        });
    }
}
