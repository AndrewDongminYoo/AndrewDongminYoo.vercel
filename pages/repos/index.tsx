import { useEffect, useRef } from 'react';
import type { GetStaticProps } from 'next';
import Head from 'next/head';
import Layout from '@components/layout';
import ReactGitHubCalendar from '@components/calendar';
import Repo from '@pages/repos/[repo]';
import type { Repository } from '@typings/repos';
import ResumeSection from '@components/section';
import { readRepositories } from '@lib/repos';
import { secondaryTitle } from '@constants';

export default function Portfolio({
    repositoryData,
}: {
    repositoryData: Repository[];
}) {
    const sectionRef = useRef<HTMLElement>(null);
    useEffect(() => {
        if (sectionRef) {
            // 홈스크린과 프로필 부분이 같으니까 스킵하기
            sectionRef.current?.firstElementChild?.scrollIntoView({
                behavior: "smooth", block: "start"
            });
        }
    }, [sectionRef])

    return (
        <Layout>
            <Head>
                <meta
                    name="keywords"
                    content="서버/백엔드, 웹 풀스택, 크로스플랫폼개발, 개발자 구인"
                />
                <title>{secondaryTitle}</title>
            </Head>
            <section
                ref={sectionRef}
                className="w-full p-6 mx-0 my-6 border border-gray-300 border-solid rounded-lg">
                {repositoryData.map((repo, id) => {
                    return <Repo repository={repo} key={id} />;
                })}
            </section>
            <ResumeSection type="contributions">
                <ReactGitHubCalendar />
            </ResumeSection>
        </Layout>
    );
}

export const getStaticProps: GetStaticProps = () => {
    const repositoryData = readRepositories();
    return { props: { repositoryData } };
};
