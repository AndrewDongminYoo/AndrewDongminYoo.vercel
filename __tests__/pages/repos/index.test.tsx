import { cleanup, render, screen } from '@testing-library/react';
import Portfolio from 'pages/repos/index';
jest.mock('lib/repos');
import { readRepositories } from 'lib/repos';

describe('Portfolio component', () => {
    afterEach(cleanup);

    it('should render correctly', () => {
        const { baseElement } = render(<Portfolio repositoryData={[]} />);
        expect(baseElement).toMatchSnapshot();
        expect(screen.getByRole('main')).toBeInTheDocument();
    });

    it('should render a section element with the correct className', () => {
        render(<Portfolio repositoryData={[]} />);
        const section = screen.getByRole('region', { name: /repositories/i });
        expect(section).toHaveClass(
            'w-full p-6 mx-0 my-6 border border-gray-300 border-solid rounded-lg'
        );
    });

    it('should render the correct number of Repo components', () => {
        const repositoryData = readRepositories();
        render(<Portfolio repositoryData={repositoryData} />);
        expect(screen.getAllByRole('heading')).toHaveLength(3);
    });
});
