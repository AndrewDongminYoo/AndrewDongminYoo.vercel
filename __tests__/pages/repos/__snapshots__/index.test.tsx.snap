// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio component should render correctly 1`] = `
<body>
    <div>
        <div
            class="mx-auto my-0 bg-white lg:max-w-4xl transition-all max-w-7xl"
        >
            <summary
                class="inline-block p-10 pb-4 overflow-x-hidden text-base xl:scrollbar-hide xl:w-120 xl:max-h-screen xl:min-h-screen xl:overflow-y-auto lg:block sm:p-4 xs:overflow-x-hidden xl:pb-16 xl:top-0"
            >
                <div
                    class="fixed z-50 flex justify-end print:hidden top-14 right-14"
                >
                    <button
                        aria-expanded="false"
                        aria-haspopup="menu"
                        class="fixed"
                        data-headlessui-state=""
                        id="headlessui-menu-button-:r0:"
                        type="button"
                    >
                        <img
                            alt="Cat Floating in Space"
                            class="relative rounded-full shadow-2xl shadow-slate-700 w-14 h-14"
                            data-nimg="1"
                            decoding="async"
                            height="56"
                            loading="lazy"
                            src="/_next/image?url=%2Fimg.jpg&w=128&q=10"
                            srcset="/_next/image?url=%2Fimg.jpg&w=64&q=10 1x, /_next/image?url=%2Fimg.jpg&w=128&q=10 2x"
                            style="color: transparent;"
                            width="56"
                        />
                        <img
                            alt="My Lovely Cat"
                            class="absolute right-0 z-10 w-14 h-14 animate-bounce motion-reduce:animate-none"
                            data-nimg="1"
                            decoding="async"
                            height="56"
                            loading="lazy"
                            src="/_next/image?url=%2Fimg.jpg&w=128&q=75"
                            srcset="/_next/image?url=%2Fimg.jpg&w=64&q=75 1x, /_next/image?url=%2Fimg.jpg&w=128&q=75 2x"
                            style="color: transparent;"
                            width="56"
                        />
                        <span
                            class="animate-ping motion-reduce:animate-none bottom-0 left-11 absolute w-3.5 h-3.5 bg-green-400 border-2 border-white dark:border-gray-800 rounded-full"
                        />
                    </button>
                </div>
                <h1
                    class="mx-0 my-4 text-2xl leading-normal writer"
                >
                    <a
                        class="writer-text"
                        href="https://github.com/AndrewDongminYoo/"
                    >
                        Dongmin, Yu
                    </a>
                </h1>
                <section
                    class="p-0 mt-0 text-base border-t-0"
                >
                    <a
                        href="https://github.com/AndrewDongminYoo/"
                    >
                        <img
                            alt="Dongmin, Yu"
                            class="rounded-full print:hidden"
                            data-nimg="1"
                            decoding="async"
                            height="156"
                            loading="lazy"
                            src="/_next/image?url=%2Fimg.jpg&w=384&q=75"
                            srcset="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=384&q=75 2x"
                            style="color: transparent;"
                            width="156"
                        />
                    </a>
                    <section
                        class="p-0 mt-0 text-base border-t-0 text-slate-800"
                    >
                        <h2
                            class="flex flex-row items-center justify-between mb-4 -ml-px text-2xl leading-snug flex-nowrap break-keep min-h-16"
                        >
                            집요하게 더 나은 답을 찾아내는 개발자 유동민입니다.
                        </h2>
                        <ul
                            class="p-0 my-0 list-none"
                        >
                            <li
                                class="flex flex-row items-center justify-start text-sm font-normal leading-relaxed tracking-normal flex-nowrap"
                            >
                                <a
                                    class="inline pb-1 text-sm font-normal leading-relaxed tracking-normal truncate"
                                    href="tel:01035661857"
                                    rel="noopener"
                                    target="_blank"
                                >
                                    <img
                                        alt="phone:01035661857"
                                        data-nimg="1"
                                        decoding="async"
                                        height="40"
                                        loading="lazy"
                                        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                                        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                                        style="color: transparent;"
                                        width="40"
                                    />
                                </a>
                            </li>
                            <li
                                class="flex flex-row items-center justify-start text-sm font-normal leading-relaxed tracking-normal flex-nowrap"
                            >
                                <a
                                    class="inline pb-1 text-sm font-normal leading-relaxed tracking-normal truncate"
                                    href="mailto:ydm2790@gmail.com"
                                    rel="noopener"
                                    target="_blank"
                                >
                                    <img
                                        alt="email:ydm2790@gmail.com"
                                        data-nimg="1"
                                        decoding="async"
                                        height="40"
                                        loading="lazy"
                                        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                                        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                                        style="color: transparent;"
                                        width="40"
                                    />
                                </a>
                            </li>
                            <li
                                class="flex flex-row items-center justify-start text-sm font-normal leading-relaxed tracking-normal flex-nowrap"
                            >
                                <a
                                    class="inline pb-1 text-sm font-normal leading-relaxed tracking-normal truncate"
                                    href="https://github.com/AndrewDongminYoo"
                                    rel="noopener"
                                    target="_blank"
                                >
                                    <img
                                        alt="github:github.com/AndrewDongminYoo"
                                        data-nimg="1"
                                        decoding="async"
                                        height="40"
                                        loading="lazy"
                                        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                                        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                                        style="color: transparent;"
                                        width="40"
                                    />
                                </a>
                            </li>
                            <li
                                class="flex flex-row items-center justify-start text-sm font-normal leading-relaxed tracking-normal flex-nowrap"
                            >
                                <a
                                    class="inline pb-1 text-sm font-normal leading-relaxed tracking-normal truncate"
                                    href="https://www.youtube.com/watch?v=vOh90eJ7VdY"
                                    rel="noopener"
                                    target="_blank"
                                >
                                    <img
                                        alt="youtube:www.youtube.com/watch?v=vOh90eJ7VdY"
                                        data-nimg="1"
                                        decoding="async"
                                        height="40"
                                        loading="lazy"
                                        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                                        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                                        style="color: transparent;"
                                        width="40"
                                    />
                                </a>
                            </li>
                        </ul>
                        <p
                            class="mt-4 text-base leading-6"
                        >
                            좋은 개발자가 되기 위해 계속해서 성장하고자 하는 집요함을 가지고 있습니다. 일상적으로 만나는 모든 문제들에 더 효율적이고 효과적인 답을 찾기 위해 끊임없이 고민하고 사유합니다. 함께 성장하는 좋은 동료가 되기 위해 노력하겠습니다.
                        </p>
                    </section>
                    <section
                        class="pt-1 mt-2 border-t-0.1 border-t-black border-solid"
                    >
                        <div
                            class="block p-0 mb-0.5 mx-0 mt-0 text-sm font-medium leading-normal"
                        >
                            <span>
                                주요 기술
                            </span>
                            <span
                                class="ml-1 text-xs leading-normal text-slate-600"
                            >
                                3
                                개
                            </span>
                        </div>
                        <ul
                            class="p-0 mt-1 mb-4 list-none"
                        >
                            <li
                                class="inline-block"
                            >
                                <div
                                    class="duration-75 ease-in-out transition-all px-1 py-0.5 my-0 ml-0 mr-1 overflow-hidden whitespace-nowrap border-0 rounded max-w-xxs bg-slate-800 text-xs font-normal text-ellipsis text-gray-50"
                                >
                                    React Native
                                </div>
                            </li>
                            <li
                                class="inline-block"
                            >
                                <div
                                    class="duration-75 ease-in-out transition-all px-1 py-0.5 my-0 ml-0 mr-1 overflow-hidden whitespace-nowrap border-0 rounded max-w-xxs bg-slate-800 text-xs font-normal text-ellipsis text-gray-50"
                                >
                                    Flutter
                                </div>
                            </li>
                            <li
                                class="inline-block"
                            >
                                <div
                                    class="duration-75 ease-in-out transition-all px-1 py-0.5 my-0 ml-0 mr-1 overflow-hidden whitespace-nowrap border-0 rounded max-w-xxs bg-slate-800 text-xs font-normal text-ellipsis text-gray-50"
                                >
                                    TS Node
                                </div>
                            </li>
                        </ul>
                        <div
                            class="block p-0 mb-0.5 mx-0 mt-0 text-sm font-medium leading-normal"
                        >
                            <span>
                                기술 태그
                            </span>
                        </div>
                        <ul
                            class="p-0 mt-1 mb-4 list-none"
                        >
                            <li
                                class="inline-block"
                            >
                                <div
                                    class="duration-75 ease-in-out transition-all px-1 py-0.5 my-0 ml-0 mr-1 max-w-xxs text-xs font-normal text-slate-800 bg-slate-200 border-0 rounded whitespace-nowrap overflow-hidden text-ellipsis"
                                >
                                    React Native (NativeModule)
                                </div>
                            </li>
                            <li
                                class="inline-block"
                            >
                                <div
                                    class="duration-75 ease-in-out transition-all px-1 py-0.5 my-0 ml-0 mr-1 max-w-xxs text-xs font-normal text-slate-800 bg-slate-200 border-0 rounded whitespace-nowrap overflow-hidden text-ellipsis"
                                >
                                    Hermes / Flipper
                                </div>
                            </li>
                            <li
                                class="inline-block"
                            >
                                <div
                                    class="duration-75 ease-in-out transition-all px-1 py-0.5 my-0 ml-0 mr-1 max-w-xxs text-xs font-normal text-slate-800 bg-slate-200 border-0 rounded whitespace-nowrap overflow-hidden text-ellipsis"
                                >
                                    Dart / Flutter
                                </div>
                            </li>
                            <li
                                class="inline-block"
                            >
                                <div
                                    class="duration-75 ease-in-out transition-all px-1 py-0.5 my-0 ml-0 mr-1 max-w-xxs text-xs font-normal text-slate-800 bg-slate-200 border-0 rounded whitespace-nowrap overflow-hidden text-ellipsis"
                                >
                                    TypeScript
                                </div>
                            </li>
                            <li
                                class="inline-block"
                            >
                                <div
                                    class="duration-75 ease-in-out transition-all px-1 py-0.5 my-0 ml-0 mr-1 max-w-xxs text-xs font-normal text-slate-800 bg-slate-200 border-0 rounded whitespace-nowrap overflow-hidden text-ellipsis"
                                >
                                    React.js / Next.js
                                </div>
                            </li>
                            <li
                                class="inline-block"
                            >
                                <div
                                    class="duration-75 ease-in-out transition-all px-1 py-0.5 my-0 ml-0 mr-1 max-w-xxs text-xs font-normal text-slate-800 bg-slate-200 border-0 rounded whitespace-nowrap overflow-hidden text-ellipsis"
                                >
                                    GitHub
                                </div>
                            </li>
                            <li
                                class="inline-block"
                            >
                                <div
                                    class="duration-75 ease-in-out transition-all px-1 py-0.5 my-0 ml-0 mr-1 max-w-xxs text-xs font-normal text-slate-800 bg-slate-200 border-0 rounded whitespace-nowrap overflow-hidden text-ellipsis"
                                >
                                    TailwindCSS
                                </div>
                            </li>
                            <li
                                class="inline-block"
                            >
                                <div
                                    class="duration-75 ease-in-out transition-all px-1 py-0.5 my-0 ml-0 mr-1 max-w-xxs text-xs font-normal text-slate-800 bg-slate-200 border-0 rounded whitespace-nowrap overflow-hidden text-ellipsis"
                                >
                                    Storybook (JS) / WidgetBook (Flutter)
                                </div>
                            </li>
                            <li
                                class="inline-block"
                            >
                                <div
                                    class="duration-75 ease-in-out transition-all px-1 py-0.5 my-0 ml-0 mr-1 max-w-xxs text-xs font-normal text-slate-800 bg-slate-200 border-0 rounded whitespace-nowrap overflow-hidden text-ellipsis"
                                >
                                    Figma / Protopie
                                </div>
                            </li>
                            <li
                                class="inline-block"
                            >
                                <div
                                    class="duration-75 ease-in-out transition-all px-1 py-0.5 my-0 ml-0 mr-1 max-w-xxs text-xs font-normal text-slate-800 bg-slate-200 border-0 rounded whitespace-nowrap overflow-hidden text-ellipsis"
                                >
                                    Firebase
                                </div>
                            </li>
                            <li
                                class="inline-block"
                            >
                                <div
                                    class="duration-75 ease-in-out transition-all px-1 py-0.5 my-0 ml-0 mr-1 max-w-xxs text-xs font-normal text-slate-800 bg-slate-200 border-0 rounded whitespace-nowrap overflow-hidden text-ellipsis"
                                >
                                    AWS CloudFront / AWS Lambda
                                </div>
                            </li>
                            <li
                                class="inline-block"
                            >
                                <div
                                    class="duration-75 ease-in-out transition-all px-1 py-0.5 my-0 ml-0 mr-1 max-w-xxs text-xs font-normal text-slate-800 bg-slate-200 border-0 rounded whitespace-nowrap overflow-hidden text-ellipsis"
                                >
                                    iOS (Swift, ObjectiveC++)
                                </div>
                            </li>
                            <li
                                class="inline-block"
                            >
                                <div
                                    class="duration-75 ease-in-out transition-all px-1 py-0.5 my-0 ml-0 mr-1 max-w-xxs text-xs font-normal text-slate-800 bg-slate-200 border-0 rounded whitespace-nowrap overflow-hidden text-ellipsis"
                                >
                                    Android (Kotlin, Gradle)
                                </div>
                            </li>
                            <li
                                class="inline-block"
                            >
                                <div
                                    class="duration-75 ease-in-out transition-all px-1 py-0.5 my-0 ml-0 mr-1 max-w-xxs text-xs font-normal text-slate-800 bg-slate-200 border-0 rounded whitespace-nowrap overflow-hidden text-ellipsis"
                                >
                                    OpenAPI / Postman
                                </div>
                            </li>
                            <li
                                class="inline-block"
                            >
                                <div
                                    class="duration-75 ease-in-out transition-all px-1 py-0.5 my-0 ml-0 mr-1 max-w-xxs text-xs font-normal text-slate-800 bg-slate-200 border-0 rounded whitespace-nowrap overflow-hidden text-ellipsis"
                                >
                                    GraphQL / Prisma / Apollo
                                </div>
                            </li>
                            <li
                                class="inline-block"
                            >
                                <div
                                    class="duration-75 ease-in-out transition-all px-1 py-0.5 my-0 ml-0 mr-1 max-w-xxs text-xs font-normal text-slate-800 bg-slate-200 border-0 rounded whitespace-nowrap overflow-hidden text-ellipsis"
                                >
                                    Selenium / BeautifulSoup
                                </div>
                            </li>
                            <li
                                class="inline-block"
                            >
                                <div
                                    class="duration-75 ease-in-out transition-all px-1 py-0.5 my-0 ml-0 mr-1 max-w-xxs text-xs font-normal text-slate-800 bg-slate-200 border-0 rounded whitespace-nowrap overflow-hidden text-ellipsis"
                                >
                                    Testing Library / React Test Renderer
                                </div>
                            </li>
                        </ul>
                    </section>
                </section>
            </summary>
            <main
                class="inline-block p-10 pt-0 overflow-x-hidden text-base xl:scrollbar-hide xl:w-full-30 xl:max-h-screen xl:min-h-screen xl:overflow-y-auto lg:block sm:p-4"
            >
                <section>
                    <section
                        aria-label="repositories"
                        class="w-full p-6 mx-0 my-6 border border-gray-300 border-solid rounded-lg"
                    />
                    <section
                        class="w-full p-6 mx-0 my-6 border border-gray-300 border-solid rounded-lg"
                        id="contributions"
                    >
                        <div
                            class="flex justify-between w-full"
                        >
                            <div
                                class="flex items-center justify-start w-32 sm:w-full max-h-8"
                            >
                                <h3
                                    class="flex items-center justify-start text-base font-medium sm:text-sm"
                                >
                                    퍼블릭 컨트리뷰션
                                </h3>
                            </div>
                            <div
                                class="flex items-center justify-end w-full-9"
                            />
                        </div>
                        <div
                            class="w-full"
                        >
                            <div
                                class="calendar"
                                title="github-calendar"
                            />
                        </div>
                    </section>
                </section>
            </main>
        </div>
    </div>
    <script
        data-nscript="afterInteractive"
        src="https://unpkg.com/github-calendar@latest/dist/github-calendar.min.js"
    />
</body>
`;
